process/smartContract/hooks/blockChainHook.go

```diff
@@ -371,7 +372,7 @@ func (bh *BlockChainHookImpl) ProcessBuiltInFunction(input *vmcommon.ContractCal
 		}
 	}
 
-	if !check.IfNil(dstAccount) {
+	if !check.IfNil(dstAccount) && !bytes.Equal(input.CallerAddr, input.RecipientAddr) {
```

Comment:
I think this will get clearer if we write it as:
```GO
shouldSaveDestination := !check.IfNil(dstAccount) && !bytes.Equal(sndAccount.AddressBytes(), dstAccount.AddressBytes)
if shouldSaveDestination{
....
```

Reply:
I would leave as it is.